lightning/src/util/logger.rs

```diff
@@ -137,8 +145,25 @@ impl<'a> Record<'a> {
 
 /// A trait encapsulating the operations required of a logger
 pub trait Logger {
-	/// Logs the `Record`
-	fn log(&self, record: &Record);
+	/// LDK users should define the log function to describe
+	/// how a log record is represented in their system
+	/// when passed an instance of the [`Record`] struct
```

Comment:
Please wrap at 100 characters.

Also, this is a bit too verbose, IMO. No need to say "when passed an instance of the [`Record`] struct" as that is implied by the method signature.

The rest is a bit vague. i.e., what are we trying to convey by saying "describe"? I think the important points to make are that the implementor is responsible to writing the record somewhere and that the record can be modified by decorators.